modules/citrus-http/src/main/java/com/consol/citrus/http/message/HttpMessage.java

```diff
@@ -416,55 +438,23 @@ public void setCookies(Cookie[] cookies) {
 
     /**
      * Adds new cookie to this http message.
-     * @param cookie
-     * @return
+     * @param cookie The Cookie to set
+     * @return The altered HttpMessage
      */
     public HttpMessage cookie(Cookie cookie) {
         this.cookies.add(cookie);
```

Comment:
Does it make sense here to overwrite cookies with same name (i.e. check using name if already in list and overwrite rather than allowing multiple cookies with the same name)?

Reply:
Well, this makes sense from my perspective as well but `cookie(Cookie cookie)` is a method exposed to the public API via a builder with the documentation of *"Adds new cookie to this http message."*.

So I'd not change this logic here. In addition, there is no technical or functional requirement for that change. Maybe this is something to consider for future major releases.

Reply:
AFAIK there is a requirement from a technical point of view (https://tools.ietf.org/html/rfc6265#section-4.1.1) 

> Servers SHOULD NOT include more than one Set-Cookie header field in
> the same response with the same cookie-name.  (See Section 5.2 for
> how user agents handle this case.)

Reply:
Okay, you've convinced me. :smile: